lib/database.js

```diff
@@ -55,24 +55,36 @@ module.exports.healthcheck = async () => {
  * @param {Object} data - Data object to be saved into the database
  * @returns {Promise} - A promise which resolves to a unique token
  */
-const insertOrUpdate = async (data) => {
+const insert = async (data) => {
   const db = await getDb();
   const collection = db.collection(COLLECTION_NAME);
-  if (data.token) {
-    const filter = { token: data.token };
-    const options = { upsert: true };
-    await collection.updateOne(filter, { $set: data }, options);
-    return data.token;
-  }
 
-  const token = uuid();
+  const token = data.token || uuid();
   await collection.insertOne({
     ...data,
     token,
   });
   return token;
 };
 
+/**
+ *
+ * @param {Object} data - Data object to be saved into the database
+ * @returns {Promise} - A promise which resolves to a unique token
+ */
+const update = async (data) => {
+  const db = await getDb();
+  const collection = db.collection(COLLECTION_NAME);
+
+  const token = data.token || uuid();
+  if (token) {
```

Comment:
Is if (token) required here? const token = data.token || uuid(); should always give token a value?

Reply:
you're right. I'll change this